lib/utils/logDeprecation.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+const chalk = require('chalk');
+const memoizee = require('memoizee');
+
+const loggedDeprecations = {};
+
+function consoleLog(...args) {
+  console.log(...args); // eslint-disable-line no-console
+}
+
+function extractCodes(codesStr) {
+  if (!codesStr) {
+    return [];
+  }
+  return codesStr.split(',').reduce((acc, code) => {
+    acc[code] = true;
+    return acc;
+  }, {});
+}
+
+const prepareCodes = memoizee((serviceConfigCodesStr, envCodesStr) => {
+  const serviceConfigCodes = extractCodes(serviceConfigCodesStr);
+  const envCodes = extractCodes(envCodesStr);
+  const disabledCodes = Object.assign({}, envCodes, serviceConfigCodes);
+  return disabledCodes;
+});
+
+function writeDeprecation(message, docURL) {
+  consoleLog(`${chalk.red(message)}\n${chalk.dim(`More Info: ${docURL}`)}`);
```

Comment:
Let's use `process.stdout.write` instead of `console.log` (it allow us to implement this without fooling linter).

Reply:
Done